Source/WebInspectorUI/UserInterface/Views/MediaDetailsSidebarPanel.js

```diff
@@ -0,0 +1,240 @@
+    supportsDOMNode(nodeToInspect)
+    {
+        return nodeToInspect.isMediaElement();
+    }
+
+    // Protected
+
+    initialLayout()
+    {
+        super.initialLayout();
+
+        this._sourceRow = new WI.DetailsSectionSimpleRow(WI.UIString("Source"));
+        this._viewportRow = new WI.DetailsSectionSimpleRow(WI.UIString("Viewport"));
+        this._framesRow = new WI.DetailsSectionSimpleRow(WI.UIString("Frames"));
+        this._resolutionRow = new WI.DetailsSectionSimpleRow(WI.UIString("Resolution"));
+        this._videoFormatRow = new WI.DetailsSectionSimpleRow(WI.UIString("Video Format"));
+        this._audioFormatRow = new WI.DetailsSectionSimpleRow(WI.UIString("Audio Format"));
+
+        let generalGroup = new WI.DetailsSectionGroup([this._sourceRow, this._viewportRow, this._framesRow, this._resolutionRow, this._videoFormatRow, this._audioFormatRow
+                                                       ]);
+        this._generalSection = new WI.DetailsSection("media-details-general", WI.UIString("General"), [generalGroup]);
+        this.element.appendChild(this._generalSection.element);
+
+        this._videoCodecRow = new WI.DetailsSectionSimpleRow(WI.UIString("Video Codec"));
+        this._transferRow = new WI.DetailsSectionSimpleRow(WI.UIString("Transfer Function"));
+        this._primariesRow = new WI.DetailsSectionSimpleRow(WI.UIString("Color Primaries"));
+        this._matrixRow = new WI.DetailsSectionSimpleRow(WI.UIString("Matrix Coefficients"));
+        this._fullRangeRow = new WI.DetailsSectionSimpleRow(WI.UIString("Color Range"));
+
+        let videoGroup = new WI.DetailsSectionGroup([this._videoCodecRow, this._primariesRow, this._transferRow, this._matrixRow, this._fullRangeRow]);
+        this._videoSection = new WI.DetailsSection("media-video-details", WI.UIString("Video Details"), [videoGroup]);
+        this.element.appendChild(this._videoSection.element);
+
+        this._audioCodecRow = new WI.DetailsSectionSimpleRow(WI.UIString("Audio Codec"));
+        this._sampleRateRow = new WI.DetailsSectionSimpleRow(WI.UIString("Sample Rate"));
+        this._channelsRow = new WI.DetailsSectionSimpleRow(WI.UIString("Channels"));
+
+        let audioGroup = new WI.DetailsSectionGroup([this._audioCodecRow, this._sampleRateRow, this._channelsRow]);
+        this._audioSection = new WI.DetailsSection("media-audio-details", WI.UIString("Audio Details"), [audioGroup]);
+        this.element.appendChild(this._audioSection.element);
+    }
+
+    async layout()
+    {
+        super.layout();
+        
+        let stats = await this.domNode.getMediaStats();
+        this.source = stats.source;
+        this.viewport = stats.viewport;
+        this.devicePixelRatio = stats.devicePixelRatio;
+        this.quality = stats.quality;
+        this.videoFormat = stats.videoFormat;
+        this.audioFormat = stats.audioFormat;
+        this.video = stats.video;
+        this.audio = stats.audio;
+    }
+
+    get source() {
```

Comment:
All of these getter/setters appear to be only used within this class, so should be prefixed with an underscore and moved below to the `// Private` section.

Reply:
Is there no convention to use actual private methods and members in WI? (E.g., `#source` rather than `_source`)

Reply:
There isn't currently because until recently there was no way in Web Inspector to see those properties or their values, so we previously didn’t use them because debugging wasn’t practical. That changed earlier this year, so IMO we could indeed start using private members and functions.

Reply:
None of those seem like good reasons to avoid private fields; if making accessors private is important, they need to be actually private, as anything else is just convention. 

It's super weird to have write-only properties; if `get foo` should be removed, then all `set foo` accessors should be turned into `setFoo(foo)` functions.